bentoml/yatai/deployment_utils.py

```diff
@@ -57,19 +57,20 @@ def deployment_dict_to_pb(deployment_dict):
 
     if deployment_pb.spec.operator == DeploymentSpec.AWS_SAGEMAKER:
         sagemaker_config = spec_dict.get('sagemaker_operator_config', {})
-        sagemaker_operator_config_pb = deployment_pb.spec.sagemaker_operator_config
+        sagemaker_config_pb = deployment_pb.spec.sagemaker_operator_config
         if sagemaker_config.get('api_name'):
-            sagemaker_operator_config_pb.api_name = sagemaker_config.get('api_name')
+            sagemaker_config_pb.api_name = sagemaker_config.get('api_name')
         if sagemaker_config.get('region'):
-            sagemaker_operator_config_pb.region = sagemaker_config.get('region')
+            sagemaker_config_pb.region = sagemaker_config.get('region')
         if sagemaker_config.get('instance_count'):
-            sagemaker_operator_config_pb.instance_count = int(
+            sagemaker_config_pb.instance_count = int(
                 sagemaker_config.get('instance_count')
             )
         if sagemaker_config.get('instance_type'):
-            sagemaker_operator_config_pb.instance_type = sagemaker_config.get(
-                'instance_type'
-            )
+            sagemaker_config_pb.instance_type = sagemaker_config.get('instance_type')
+        if sagemaker_config.get('num_of_gunicorn_workers_per_instance'):
+            sagemaker_config_pb.num_of_gunicorn_workers_per_instance = \
+                sagemaker_config.get('num_of_gunicorn_workers_per_instance')
```

Comment:
let's add a unit test for `deployment_dict_to_pb`? it seems easy to forget about it when adding new field in proto message

Reply:
We do have one basic test for `deployment_dict_to_pb`.  Yes, I agree we need to add more of it